src/praisonai/praisonai/bots/linear.py

```
@@ -0,0 +1,504 @@
+            raise ValueError("LINEAR_OAUTH_TOKEN or LINEAR_API_KEY required")
+            
+        if not self._signing_secret:
+            logger.warning("LINEAR_WEBHOOK_SECRET not set - webhook signatures will not be verified")
+
+        logger.info(f"Starting Linear bot on port {self._webhook_port}")
+        
+        try:
+            import aiohttp
+            from aiohttp import web
+        except ImportError:
+            raise ImportError("aiohttp required: pip install aiohttp")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)